File: Swift/프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.swift
import Foundation

func solution(_ arr:[Int]) -> Int {
    var arr = arr
    var count = 0
    
    while true {
        var temp: [Int] = []
        
        for num in arr {
             if num >= 50 && num % 2 == 0 {
                 temp.append(num / 2)
             }
            else if num < 50 && num % 2 != 0 {
                temp.append(num * 2 + 1)
            }  
            else {
                temp.append(num)
            }       
        }
           
        if temp == arr {
            return count
        }
         
        arr = temp
        count += 1
    }
        
}

File: Python3/프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):    
    count = 0
    
    while True:
        temp = []
        
        for num in arr:
            if num >= 50 and num % 2 == 0:
                temp.append(num // 2)
            elif num < 50 and num % 2 != 0:
                temp.append(num * 2 + 1)
            else:
                temp.append(num)
        
        
        if temp == arr:
            return count
        
        arr = temp
        count += 1
        